lib/npm-auditer.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/* eslint-disable max-len */
+const childProcess = require('child_process');
+
+function reportAudit(report, config) {
+  return new Promise((resolve, reject) => {
+    if (config.report) {
+      console.log('\x1b[36m%s\x1b[0m', 'NPM audit report JSON:');
+      console.log(JSON.stringify(report, null, 2));
+    }
+
+    const { levels, whitelist } = config;
+
+    if (whitelist.length) {
+      console.log(`Modules to whitelist: ${whitelist.join(', ')}.`);
+    }
+    const { vulnerabilities } = report.metadata;
+    const failedLevels = [];
+    const whitelistedFound = [];
+    let advisories = null;
+    ['low', 'moderate', 'high', 'critical'].forEach(level => {
+      if (vulnerabilities[level] > 0 && levels[level]) {
+        if (whitelist.length) {
+          // lazily intiialize advisories for max efficiency
+          if (!advisories) {
+            advisories = Object.values(report.advisories);
+          }
+          advisories.forEach(advisory => {
+            if (
+              advisory.severity === level &&
+              whitelist.some(m => m === advisory.module_name)
+            ) {
+              whitelistedFound.push(advisory.module_name);
+            } else {
+              failedLevels.push(level);
+            }
+          });
+        } else {
+          failedLevels.push(level);
+        }
+      }
+    });
+
+    if (whitelistedFound.length) {
+      const found = whitelistedFound.join(', ');
+      const msg = `Vulnerable whitelisted modules found: ${found}.`;
+      console.warn(msg);
+    }
+
+    if (failedLevels.length) {
+      const failedLevelsFound = failedLevels.join(', ');
+      const err = `Failed security audit due to ${failedLevelsFound} vulnerabilities.`;
+      reject(Error(err));
+    } else {
+      resolve(report);
+    }
+  });
+}
+
+function runNpmAudit() {
+  return new Promise((resolve, reject) => {
+    childProcess.exec('npm audit --json', (_error, stdout, stderr) => {
+      if (stderr) {
+        reject(stderr);
+        return;
+      }
+      const parsedAudit = JSON.parse(stdout);
+      resolve(parsedAudit);
+    });
+  });
+}
+
+/**
+ * Audit your NPM project!
+ *
+ * @param {{ report: boolean, whitelist: string[], levels: { low: boolean, moderate: boolean, high: boolean, critical: boolean }}} config
+ * `report`: whether to show the NPM audit report in the console.
+ * `whitelist`: a list of packages that should not break the build if their vulnerability is found.
+ * `levels`: whether the audit should fail if found at that level.
```

Comment:
The JSDoc was updated according to the expectation of using the `levels` arg attached at run-time in `audit-ci.js`